File: python-sdk/src/boxsdk/models/operations/post_files_id_versions_current.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import fileversion as shared_fileversion


@dataclasses.dataclass
class PostFilesIDVersionsCurrentPathParams:
    file_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'file_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class PostFilesIDVersionsCurrentQueryParams:
    fields: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'fields', 'style': 'form', 'explode': False }})
    
class PostFilesIDVersionsCurrentRequestBodyTypeEnum(str, Enum):
    FILE_VERSION = "file_version"


@dataclass_json
@dataclasses.dataclass
class PostFilesIDVersionsCurrentRequestBody:
    r"""PostFilesIDVersionsCurrentRequestBody
    The file version to promote
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: Optional[PostFilesIDVersionsCurrentRequestBodyTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclasses.dataclass
class PostFilesIDVersionsCurrentRequest:
    path_params: PostFilesIDVersionsCurrentPathParams = dataclasses.field()
    query_params: PostFilesIDVersionsCurrentQueryParams = dataclasses.field()
    request: Optional[PostFilesIDVersionsCurrentRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostFilesIDVersionsCurrentResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    file_version: Optional[shared_fileversion.FileVersion] = dataclasses.field(default=None)
    

File: ts-sdk/src/sdk/models/operations/postfilesidversionscurrent.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostFilesIdVersionsCurrentPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class PostFilesIdVersionsCurrentQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}

export enum PostFilesIdVersionsCurrentRequestBodyTypeEnum {
    FileVersion = "file_version"
}


// PostFilesIdVersionsCurrentRequestBody
/** 
 * The file version to promote
**/
export class PostFilesIdVersionsCurrentRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: PostFilesIdVersionsCurrentRequestBodyTypeEnum;
}


export class PostFilesIdVersionsCurrentRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostFilesIdVersionsCurrentPathParams;

  @SpeakeasyMetadata()
  queryParams: PostFilesIdVersionsCurrentQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostFilesIdVersionsCurrentRequestBody;
}


export class PostFilesIdVersionsCurrentResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileVersion?: shared.FileVersion;

  @SpeakeasyMetadata()
  statusCode: number;
}
